uswds/uswds
# USWDS: Button - Forced colors outline

## Summary
This PR adds an outline to buttons in forced color mode for added clarity and consistency with default styles. It also replaces repetitive CSS with a new `$border-high-contrast` variable

## Breaking change
This is **not** a breaking change

## Related issue
Closes #5099 

## Preview link
[Buttons Preview →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-button-forced-colors-outline/?path=/story/components-button--default)

## Problem statement
Currently, buttons in forced-colors modes do not have an outline or any styling to appear like buttons. Instead, it just looks like regular text unless focused.

## Solution
Define an outline within forced colors mode. Other components utilize similar styling in forced colors mode, so I've created a mixin and included in on `usa-button`

Previously we set high-contrast outlines via `outline: 2px solid transparent` because it would not affect the border outside of high-contrast modes. Since we are using the `forced-colors` media query for all high contrast styles, we can adjust the border without affecting components in normal circustances. Thus, high contrast borders have been turned into a sass variable and made uniform across our components


## Testing and review
1. View the Button story preview
2. In developer tools, click the 3-dot menu button and select "Rendering" under more tools
3. Scroll down to **Emulate css media feature forced-colors** and select `Forced-colors: active"
4. Verify that buttons have an outline making them appear as buttons
5. Compare Checkbox, Radio, Search, Range, and Date Picker components to the develop build in high contrast mode to ensure no style change as a result of switching to outline mixin.

Before opening this PR, make sure you’ve done whichever of these applies to you:
